### PR TITLE
Add support for clang + move to git 2.40.1

### PR DESCRIPTION
**Pull request by IgorTodorovskiIBM**
The CC_LD_DYNPATH is a bit of a hack because if CC_LD_DYNPATH is set to null, then the directory is added to the link step, leading to a linker error when a dir is specified. This linker error does not occur with xlclang when a dir is specified. It does occur with clang on other platforms though, but the other platforms do not have CC_LD_DYNPATH set to null.

In addition, clang is a bit more strict, so validate_codeset had to be changed to a void return type since the returns return no value.

### CODE REVIEW

**Comment by IgorTodorovskiIBM** [comment]
Also added in this PR is the move up to git-2.40.1. Additionally, I have exposed git-jump utility which is documented here: https://github.blog/2023-03-13-highlights-from-git-2-40/

File: tarballpatches/config.mak.uname.patch [hunk 1]
```diff
@@ -1,28 +1,17 @@
...
  	SHELL_PATH = /usr/coreutils/bin/bash
  endif
-+
 +ifeq ($(uname_S),OS/390)
-+  V = 2
-+	PERL_PATH = perl
-+	PERL_PATH_FOR_SCRIPTS = /bin/env perl
-+	SHELL_PATH = bash
-+	SHELL_PATH_FOR_SCRIPTS = /bin/env bash
-+	PYTHON_PATH = python
```
Comment (MikeFultonDev) [inline tarballpatches/config.mak.uname.patch:25]: is python a dependency for git test ? I don't see it in the buildenv

**Review by MikeFultonDev — COMMENTED** [review]
just a comment on python

File: tarballpatches/config.mak.uname.patch (hunk 1 above)
Comment (IgorTodorovskiIBM) [reply tarballpatches/config.mak.uname.patch:25]: Thanks, opened issue https://github.com/ZOSOpenTools/gitport/issues/68.

**Review by MikeFultonDev — APPROVED** [review]

**Pull request merged**